### ISSUE TITLE
<auto-generated> Comment Changes Diagnostics

### ISSUE THREAD

**Issue by TanayParikh** [dotnet/roslyn#48455]
**Version Used**: VS Preview 16.8-Preview4

**Steps to Reproduce**:

1. Create a `Counter.cs` file as part of a project.
2. Paste in:

```csharp
// <auto-generated>
namespace blazorserver
{
    using System.Diagnostics;

    public class Counter2
    {
        private void SomethingLaunching()
        {
            System.Diagnostics.Debugger.Launch();
        }
    }
}
```

3. Note, the `System.Diagnostics` FQN is in bright white (it takes precedence over `using System.Diagnostics`.
4. Remove `<auto-generated`. Note the FQN is in a lighter color (signifying it can be simplified `IDE0002`)

**Expected Behavior**: The `<auto-generated>` comment should not change the behavior.
![image](https://user-images.githubusercontent.com/14852843/95510702-32997300-096b-11eb-8cea-63c4c44e65c5.png)

**Actual Behavior**: The presence of the `<auto-generated>` comment determines whether the `using` statement or the FQN type takes precedence.
![image](https://user-images.githubusercontent.com/14852843/95510740-3d540800-096b-11eb-87a2-5753484eb331.png)

![Autogenerated](https://user-images.githubusercontent.com/14852843/95510629-1564a480-096b-11eb-8faa-2f471cbe38df.gif)

**Comment by Youssef1313** [comment]
I believe this design was intentional, most IDExxxx (and analyzers in general - except security-related analyzers) are configured not to work on generated code:

https://github.com/dotnet/roslyn/blob/cf55f3a58e47298426fa971d3bd9d8857c746c65/src/Features/Core/Portable/SimplifyTypeNames/SimplifyTypeNamesDiagnosticAnalyzerBase.cs#L93

If the analyzer reports on the code, then you apply the codefix, and this file gets auto-generated again, you'll just end in this loop.
The fix should be done in whatever auto-generates this code.

**Comment by TanayParikh** [comment]
>After reading again, I think you want IDE0002 to trigger in the .razor file (which doesn't have the auto-generated comment, but it affects the auto-generated code)? Please clarify if that's the case.

Yes. Ideally, `IDE0002` would be present in the `.razor` file as well. From my understanding, razor gets these diagnostics from the underlying `.razor.g.cs` file which is prefixed with the `<auto-generated>` comment. 

> If the analyzer reports on the code, then you apply the codefix, and this file gets auto-generated again, you'll just end in this loop.

I don't believe this would be an issue in the razor case. Once the analyzer reports a particular diagnostic, and a code action is applied in the `.razor` file, the generated code would be updated with the applied code fix (no-loop).

>most IDExxxx (and analyzers in general - except security-related analyzers) are configured not to work on generated code:

I think generally it may make sense to toggle on support for `IDExxxx` / analyzers for razor generated files so that those diagnostics are available in turn in the `.razor` file. Would this be a potential configurable option?

cc/ @NTaylorMullen @ajaybhargavb

**Comment by Youssef1313** [comment]
@TanayParikh I have no idea how the compiler works with `.razor` files. But it sounds reasonable that `.razor` file gets IDE diagnostics, but *not* generated C# file.

**Comment by sharwell** [comment]
Simplify type names does not run on generated code because it's a _very_ expensive analyzer and generated code is often _very_ large. Even if we don't show suggestions to make changes, we would have to run the analyzer to identify locations to fade.

**Comment by sharwell** [comment]
@TanayParikh we should come up with a new title for this. The original question is By Design, but it seems like the actual question you're looking for an answer to involves a _much_ more specific scenario.
